sequencing-server/src/utils/hasura.ts

```diff
@@ -0,0 +1,314 @@
+      return (
+        (planId !== null && (await isPlanOwner(graphqlClient, username, planId))) ||
+        (await isPlanCollaborator(graphqlClient, username, planId))
+      );
+  }
+}
+
+async function isMissionModelOwner(
+  graphqlClient: GraphQLClient,
+  username: string,
+  missionModelId: number,
+): Promise<boolean> {
+  const missionModelOwner = await graphqlClient.request<{
+    mission_model_by_pk: { owner: string };
+  }>(
+    gql`
+      query missionModelOwner($missionModelId: Int!) {
+        mission_model_by_pk(id: $missionModelId) {
+          owner
+        }
+      }
+    `,
+    { missionModelId },
+  );
+
+  return username === missionModelOwner.mission_model_by_pk?.owner;
+}
+
+async function isPlanOwner(graphqlClient: GraphQLClient, username: string, planId: number): Promise<boolean> {
+  const planOwner = await graphqlClient.request<{
+    plan_by_pk: { owner: string };
+  }>(
+    gql`
+      query planOwner($planId: Int!) {
+        plan_by_pk(id $planId) {
+          owner
+        }
+      }
+    `,
+    { planId },
+  );
+
+  return username === planOwner.plan_by_pk?.owner;
+}
+
+async function isPlanCollaborator(graphqlClient: GraphQLClient, username: string, planId: number): Promise<boolean> {
+  const planCollaborator = await graphqlClient.request<{
+    plan_collaborators_by_pk: { collaborator: string | null };
+  }>(
+    gql`
+      query planCollaborator($planId: Int!, $username: String!) {
+        plan_collaborators_by_pk(plan_id: $planId, collaborator: $username) {
+          collaborator
+        }
+      }
+    `,
+    { planId, username },
+  );
+
+  return planCollaborator !== null;
+}
+
+async function getPlanId(
+  graphqlClient: GraphQLClient,
+  missionModelId?: number,
+  simulationDatasetId?: number,
+): Promise<number> {
+  let planId;
+
+  if (missionModelId) {
+    planId = (
+      await graphqlClient.request<{
+        plan: { id: number }[];
+      }>(
+        gql`
+          query plan($missionModelId: Int!) {
+            plan(where: { model_id: { _eq: $missionModelId } }) {
+              id
+            }
+          }
+        `,
+        { missionModelId },
+      )
+    ).plan[0]?.id;
+  } else if (simulationDatasetId) {
+    planId = (
+      await graphqlClient.request<{
+        plan: { id: number }[];
+      }>(
+        gql`
+          query plan($simulationDatasetId: Int!) {
+            plan(where: { simulations: { simulation_dataset: { id: { _eq: $simulationDatasetId } } } }) {
+              id
+            }
+          }
+        `,
+        { simulationDatasetId },
+      )
+    ).plan[0]?.id;
+  }
+
+  if (planId === null || planId === undefined) {
+    throw new Error('Could not determine the plan based on the provided missionModelId or simulationDatasetId');
+  }
+
+  return planId;
+}
+
+async function getMissionModelId(
+  graphqlClient: GraphQLClient,
+  planId?: number,
+  simulationDatasetId?: number,
+): Promise<number> {
+  if (planId) {
+    planId = (
```

Comment:
I believe this should be `missionModelId`? In general, it looks like this method is calling `missionModelId` `planId`